### ISSUE TITLE
Inconsistent handling of file paths in databasechangelog as of 4.18.x

### ISSUE THREAD

**Issue by aegliv** [liquibase/liquibase#3798]
## Environment

**Liquibase Version**: 4.19.0

**Liquibase Integration & Version**: Spring Boot 3.0.2

**Liquibase Extension(s) & Version**: n/a

**Database Vendor & Version**: PostgreSQL 14.5, compiled by Visual C++ build 1914, 64-bit startet

**Operating System Type & Version**: Win 10

**Infrastructure Type/Provider**: <AWC, GCS, Azure, VM, etc> na

## Description

Similar to https://github.com/liquibase/liquibase/issues/3028, file paths are inconsistently handled in regards to leading slashes.

## Steps To Reproduce

Execute a rollback to tag on recently applied change set

## Actual Behavior

Excecuting a changelog registers the log **with a leading slash**

```
2023-02-09T09:11:43.024+01:00  INFO 12356 --- [           main] liquibase.changelog                      : ChangeSet /db/002.yaml::4::tester ran successfully in 6ms
2023-02-09T09:11:43.024+01:00 DEBUG 12356 --- [           main] liquibase.executor                       : Executing with the 'jdbc' executor
2023-02-09T09:11:43.025+01:00 DEBUG 12356 --- [           main] liquibase.executor                       : INSERT INTO public.databasechangelog (ID, AUTHOR, FILENAME, DATEEXECUTED, ORDEREXECUTED, MD5SUM, DESCRIPTION, COMMENTS, EXECTYPE, CONTEXTS, LABELS, LIQUIBASE, DEPLOYMENT_ID) VALUES ('4', 'tester', '/db/002.yaml', NOW(), 5, '8:8082054c540ac946dbd1b206c17d6008', 'sql', '', 'EXECUTED', NULL, NULL, '4.19.0', '5930302862')
2023-02-09T09:11:43.025+01:00 DEBUG 12356 --- [           main] liquibase.executor                       : 1 row(s) affected
```

But rollback is trying to delte the row **without leading slash**

```
2023-02-09T09:11:43.699+01:00 DEBUG 12356 --- [           main] liquibase.executor                       : Rolling Back ChangeSet: db/002.yaml::4::tester
2023-02-09T09:11:43.699+01:00 DEBUG 12356 --- [           main] liquibase.executor                       : Executing with the 'jdbc' executor
2023-02-09T09:11:43.700+01:00 DEBUG 12356 --- [           main] liquibase.executor                       : DELETE FROM public.databasechangelog WHERE ID = '4' AND AUTHOR = 'tester' AND FILENAME = 'db/002.yaml'
2023-02-09T09:11:43.700+01:00 DEBUG 12356 --- [           main] liquibase.executor                       : 0 row(s) affected
```

## Expected/Desired Behavior
Revert back to the old behavior in 4.17.2, keeping the leading slash

**Comment by aegliv** [comment]
@kataggart, @nvoxland anyone plz?

**Comment by FBurguer** [comment]
Hello! Can you explain to me how did you get a '/' before the chaneglog path and why you need it? Thanks!

**Comment by aegliv** [comment]
Our changelog masterlooks like this:

````yml
databaseChangeLog:
  - include:
      file: /db/001_ddl.yaml
  - include:
      file: /db/002_copy-locktable.yaml
````

This worked fine until 4.18.x, because now these file paths are treated differently. If I replace `/db` with only `db` (no leading slash) the rollback will work fine *on a previously empty db*, but since we already have many successfull migrations in our db, and liquibase used to store the *whole path* in the `databsaechangelog` table, this behavior change is a breaking change for all migrations which ran prior to 4.18.x!

**Comment by FBurguer** [comment]
Hello! Thanks for the response! We did find the PR that broke this, this was on 4.17.x. We will look into a fix for rthe next release, but if you're interested in helping it could get done a bit sooner.

**Comment by aegliv** [comment]
thx for the fix, issue is resolved with 4.21.0! 🥳
